### PR TITLE
feat(clients): add generate_secured_api_key to ruby

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1777](https://algolia.atlassian.net/browse/DI-1777)

Add `generate_secured_api_key` and `get_secured_api_key_remaining_validity` for ruby

[DI-1777]: https://algolia.atlassian.net/browse/DI-1777?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/ruby/search_helpers.mustache
Comment (shortcuts) [inline templates/ruby/search_helpers.mustache:1]: could you add some base tests like https://github.com/algolia/api-clients-automation/blob/3f692ae3df6dd2807d25c68ed0f57b158937cf3c/tests/output/python/tests/requests/search_test.py#L3390?

**Review by shortcuts — COMMENTED** [review]
looks great!

File: templates/ruby/search_helpers.mustache
Comment (millotp) [reply templates/ruby/search_helpers.mustache:1]: I tried it in #2798 actually with generated tests ! but only 4/11 lang are ready so I can't merge this PR yet

Comment (shortcuts) [reply templates/ruby/search_helpers.mustache:1]: ah oki I didn't thought you would continue it before ga

**Review by shortcuts — APPROVED** [review]
noice

**Pull request merged**